engine/graphics/src/graphics_adapter.h

```diff
@@ -100,11 +100,13 @@ namespace dmGraphics
     typedef bool (*ReloadFragmentProgramFn)(HFragmentProgram prog, ShaderDesc::Shader* ddf);
     typedef void (*DeleteVertexProgramFn)(HVertexProgram prog);
     typedef void (*DeleteFragmentProgramFn)(HFragmentProgram prog);
+    typedef ShaderDesc::Language (*GetShaderProgramLanguageFn)(HContext context, ShaderDesc::ShaderClass shader_class);
     typedef ShaderDesc::Language (*GetProgramLanguageFn)(HProgram program);
-    typedef ShaderDesc::Language (*GetShaderProgramLanguageFn)(HContext context);
     typedef void (*EnableProgramFn)(HContext context, HProgram program);
     typedef void (*DisableProgramFn)(HContext context);
-    typedef bool (*ReloadProgramFn)(HContext context, HProgram program, HVertexProgram vert_program, HFragmentProgram frag_program);
+    typedef bool (*ReloadProgramGraphicsFn)(HContext context, HProgram program, HVertexProgram vert_program, HFragmentProgram frag_program);
+    typedef bool (*ReloadComputeProgramFn)(HComputeProgram prog, ShaderDesc::Shader* ddf);
+    typedef bool (*ReloadProgramComputeFn)(HContext context, HProgram program, HComputeProgram compute_program);
```

Comment:
In all of the graphics adapters there is actually almost no distinction between how we create the shader modules, so we are now adding more and more functions that do the same work under the hood.. maybe we should try to merge some of the API into fewer functions? 

If I could do it the way I want to, I would remove HFragmentProgram, HVertexProgram and HComputeProgram, and just represent them as a "HShader" or "HShaderModule" or something like that.